plugins/out_s3/s3.c

```diff
@@ -652,9 +671,16 @@ static int cb_s3_init(struct flb_output_instance *ins,
             flb_plg_error(ctx->ins, "upload_chunk_size must be at least 5,242,880 bytes");
             return -1;
         }
-        if (ctx->upload_chunk_size > MAX_CHUNKED_UPLOAD_SIZE) {
-            flb_plg_error(ctx->ins, "Max upload_chunk_size is 50M");
-            return -1;
+        if (ctx->compression == FLB_AWS_COMPRESS_GZIP) {
+            if(ctx->upload_chunk_size > MAX_CHUNKED_UPLOAD_COMPRESS_SIZE) {
+                flb_plg_error(ctx->ins, "upload_chunk_size in compress multipart upload cannot over 5GB");
```

Comment:
language nit: "upload_chunk_size in compress multipart upload cannot exceed 5GB"

Reply:
will do